### PR TITLE
Minor settings screen UI updates

### LINKED ISSUE #5018: Minor UI updates on settings screen
## Feature description

Copying from Slack.

1.
> Can you reduce the bottom margin on the headings? It's 1rem on bottom but 0 on the top:
image (1)

2. 
> There's also excessive margin at the bottom of toggles:

image (2)

3. 
> The approach to highlight selected boxes, could be made so that the gray area does not show when boxes are not selected. Boxes that are not "selectable" would not have the gray side, and only the one selected will have the blue side.  Would that make the UI sleeker?

4. 
> The title of the settings screen should be bold (?) and there should be a bit more space from the title to the first box:

5.
> WDYT about a bit less space between the boxes? [in the mode selection section]

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### LINKED ISSUE #5043: Prevent showing admin settings UI until REST API requests have finished loading
## Bug Description

I noticed when accessing the admin screen, there is a short period of time where none of the fields are populated:

image

Only then to be followed up with the previously-selected options being indicated:

image

## Expected Behaviour

To improve layout stability and reduce anxiety about settings not being persisted, the entire UI should either be hidden or replaced with a loading indicator while waiting for the REST API requests to complete.

## Steps to reproduce

1. Navigate to the AMP settings screen.
2. Select Standard mode and save.
3. Do a hard-reload of the page and notice the Standard mode is initially unselected.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by johnwatkins0**
~This depends on #5044~ ✅ 

## Summary

Fixes #5018
Fixes #5043

See changes below:

### Required
- [x] "Can you reduce the bottom margin on the headings? It's 1rem on bottom but 0 on the top" Done in edd3c11
- [x] "There's also excessive margin at the bottom of toggles:" Done in 2e6ddfe59
- [x] "The title of the settings screen should be bold (?) and there should be a bit more space from the title to the first box:" d3b73f054
- [x] "WDYT about a bit less space between the boxes? [in the mode selection section]" d3b73f054

### Maybe
- [x] "explore placing the plugin suppression and template sections into an Advanced panel"
- [ ]  "The approach to highlight selected boxes, could be made so that the gray area does not show when boxes are not selected. Boxes that are not "selectable" would not have the gray side, and only the one selected will have the blue side. Would that make the UI sleeker?"  _**This is probably obsolete with upcoming style adjustments**_

### Also included

- Prevent showing settings UI while options are being fetched
- Made style updates applicable to earlier WP 5.x versions
- Set max width on main content area of settings page
- Made spacing adjustments on the settings screen
- Added a customizer link to the Welcome section of the settings screen
- Made input styles consistent
- Went through CSS tried to combine and remove lines where possible
- Placed the mobile redirection toggle inside the advanced panel
- Placed the Reader themes inside a panel (to be replaced soon with the carousel)
- I **removed some e2e tests** that do not pass reliably locally. I got heavily into e2e tests a few weeks back on another project and began implementing more of them here in my PRs, but their pitfalls are already becoming obvious; it becomes impossible to run them locally, and debugging local-only failures is a huge hassle. I think with our current environment we need to keep e2e test cases confined to small and discrete scenarios that avoid many external http requests or page navigations. I'm sure a more robust way of running e2e tests across environments is possible, though.

## Checklist

- [x] My pull request is addressing an open issue (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by johnwatkins0** [comment]
@jwold If you have time Thursday morning the production build above could use a look.

File: src/Admin/OptionsMenu.php
Comment (westonruter) [inline src/Admin/OptionsMenu.php]: I don't think this is quite right. Given that the Customizer URL varies by the mode, I think the Customizer URL needs to be returned as a `readonly` property from the `options` endpoint in the same way that the `preview_permalink` is being returned. Then this link can be conditionally showed based on whether the the Options endpoint returns this URL as being non-empty and it can be updated when the template mode and reader theme are changed.

File: includes/admin/functions.php
Comment (westonruter) [inline includes/admin/functions.php]: This function should also return empty if the mode is Reader, or else it should prevent adding any `$args` other than the `url`.

File: src/Admin/OptionsMenu.php
Comment (westonruter) [reply src/Admin/OptionsMenu.php]: Oh, then I see that will depend on moving this whole `settings-welcome` into being part of the React app as well.

**Review by westonruter — CHANGES_REQUESTED** [review]
> Placed the Reader themes inside a panel (to be replaced soon with the carousel)

I think this should be rendered as part of the Reader template mode, perhaps directly connected to the the box. That would seem more natural perhaps? This also relates to having the templates mode boxes able to be collapsed. If the Reader template mode could be collapsed, could there be a another collapsable section attached to the bottom of the Reader mode section which appears when Reader mode is selected? Maybe that attached section needn't even be collapsed if it is a carousel. If it is collapsed, it should indicate the theme currently chosen rather than just “Choose Reader Theme”. If the carousel were used, then the selected theme would be indicated by being visually selected.

I currently see:

image

File: includes/admin/functions.php
Comment (johnwatkins0) [reply includes/admin/functions.php]: Done in 3085b3fc8

File: src/Admin/OptionsMenu.php
Comment (johnwatkins0) [reply src/Admin/OptionsMenu.php]: Done in 3085b3fc8

File: assets/src/settings-page/welcome.js
Comment (amedina) [inline assets/src/settings-page/welcome.js:95]: When in the middle of making changes in the Settings screen, should clicking on `Visit Customizer` link warn the user that changes have been made which have not been saved? The user can select e.g. 2019, then visit customizer to make changes on currently selected theme, then go back and save the. settings changes.

**Review by amedina — COMMENTED** [review]
We should look at the proportions between the headings and subheadings in the Settings screen. Can be done after beta1. 

Example:
Screen Shot 2020-07-16 at 7 52 42 AM

Example:
Screen Shot 2020-07-16 at 7 57 33 AM

**Comment by johnwatkins0** [comment]
> > Placed the Reader themes inside a panel (to be replaced soon with the carousel)
> 
> I think this should be rendered as part of the Reader template mode, perhaps directly connected to the the box. 

Did this in 1fa3d8837

Screen Shot 2020-07-16 at 10 06 11 AM

File: assets/src/settings-page/welcome.js
Comment (johnwatkins0) [reply assets/src/settings-page/welcome.js:95]: We could do this very easily by having the customizer link open in the same tab. Then users would see the "Leave site?" warning if they have unsaved changes.

File: includes/admin/functions.php
Comment (johnwatkins0) [reply includes/admin/functions.php]: See 141249663 for the actual fix, discussed in our call.

File: assets/src/settings-page/welcome.js
Comment (westonruter) [reply assets/src/settings-page/welcome.js:95]: Yes, let's do this. We need to warn the user because if they switch the Reader theme or mode then the Customizer link may not be current. So they need to save so that the up-to-date link is shown.

File: assets/src/settings-page/welcome.js
Comment (westonruter) [inline assets/src/settings-page/welcome.js]:
```suggestion
							<a className="components-button is-link" href={ customizerLink } rel="noreferrer">
```

File: assets/src/settings-page/welcome.js
Comment (westonruter) [reply assets/src/settings-page/welcome.js:95]: Done in 13193a5

File: src/OptionsRESTController.php
Comment (westonruter) [inline src/OptionsRESTController.php]: @johnwatkins0 Why was this change made? It broke `OptionsRESTControllerTest::test_update_items()` and prevented the `enum` constraints and other schema requirements from being enforced from the `POST` request. 

I'm reverting for now.

Comment (westonruter) [reply src/OptionsRESTController.php]: I see it was needed for WP 5.0 compat in 871bd100c040f5bbc63456a75c5e101a00434187.

I'll conditionally make this change then if the WP version is less than 5.1

Comment (johnwatkins0) [reply src/OptionsRESTController.php]: Ah, right, I meant to revisit that with a `method_exists` check or similar.

Comment (westonruter) [reply src/OptionsRESTController.php]: Actually, I see no REST API error when this method is present in 5.0. Was it for 4.9-compat?

Comment (johnwatkins0) [reply src/OptionsRESTController.php]: It came up when I was in 5.0, but I see `@since 4.7` on that method, so I must have been mistaken about whatever error I was seeing.

File: assets/src/settings-page/index.js
Comment (westonruter) [inline assets/src/settings-page/index.js:103]: Perfect

File: assets/src/settings-page/plugin-suppression.js
Comment (westonruter) [inline assets/src/settings-page/plugin-suppression.js:298]: We don't need `fetchingOptions` in `supported-templates.js` either now I presume.

Comment (westonruter) [reply assets/src/settings-page/plugin-suppression.js:298]: Removed in 35e5fbe5d

**Review by westonruter — APPROVED** [review]
Great work! Solid iteration on the screen. 

I'd like to get a second +1 from @pierlon before merging.

**Review by amedina — APPROVED** [review]
Pending additional +1 from Pierre.

File: assets/src/components/reader-theme-selection/index.js
Comment (pierlon) [inline assets/src/components/reader-theme-selection/index.js]:
```suggestion
								/* translators: placeholder is the name of a WordPress theme. */
```

File: tests/e2e/specs/amp-onboarding/template-mode-revisit.js
Comment (pierlon) [inline tests/e2e/specs/amp-onboarding/template-mode-revisit.js:1]: Was deleting this intentional, or are the tests being handled somewhere else?

**Review by pierlon — CHANGES_REQUESTED** [review]

File: tests/e2e/specs/amp-onboarding/template-mode-revisit.js
Comment (westonruter) [reply tests/e2e/specs/amp-onboarding/template-mode-revisit.js:1]: See the PR description. I think it was intentional.

Comment (johnwatkins0) [reply tests/e2e/specs/amp-onboarding/template-mode-revisit.js:1]: It was intentional because they were brittle and were bound to cause headaches at some point. They're trying to do too much. I learned (or had the lesson reinforced) from these tests that e2e tests should go to a page, do something, test the result, and be done. Having multiple page navigations within a test was turning out to be problematic. I'll work on hardening the tests via unit testing and more discrete e2e tests in the next phase of work.

Comment (pierlon) [reply tests/e2e/specs/amp-onboarding/template-mode-revisit.js:1]: Ah OK, thanks for the insight :+1:.

**Review by pierlon — APPROVED** [review]
Well done!

**Pull request merged**
